mautic/mautic
# Add support days-old for gdpr maintenance command

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR added support for `--days-old` If you use flag `--gdpr`
By default it's still 3*365 days, but you can configure it now by --days-old flag




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create contact  and modify hist  created date more like 3 years
3. Run `php /path/to/mautic/bin/console mautic:maintenance:cleanup --gdpr --days-old=1460` (expect less than 4 years old)
4. Should not delete contact
5.  Run `php /path/to/mautic/bin/console mautic:maintenance:cleanup --gdpr` 
6. Should delete contact
